examples/chaiscript/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+cmake_minimum_required(VERSION 3.2)
+
+# Emulate HunterGate:
+# * https://github.com/hunter-packages/gate
+include("../common.cmake")
+
+project(download-ChaiScript)
+
+# DOCUMENTATION_START {
+hunter_add_package(chaiscript)
+find_package(chaiscript CONFIG REQUIRED)
+# DOCUMENTATION_END }
+
+if (CMAKE_HOST_WIN32 AND (CYGWIN OR MINGW)) 
+    # set O3 to avoid a bigObj issue
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3")
+endif()
+
+get_property(chaiscript_so_path TARGET chaiscript::chaiscript PROPERTY LOCATION)
+message("Chaiscript PATH:  ${chaiscript_so_path}")
+
+add_executable(chaitest main.cpp)
+target_link_libraries(chaitest chaiscript::chaiscript) #parser chaiscript::stdlib chaiscript::stl_extra)
+
+enable_testing()
+file(COPY ${chaiscript_so_path} DESTINATION ${CMAKE_BINARY_DIR})
```

Comment:
This kind of trick will not work for multi-configuration generators. What we are trying to achieve here?

Reply:
I'm trying to figure out how to add the chaiscript .so's to the path - they do not seem to be added, and this may be an issue with how I built the Chaiscript project (not properly exported maybe).

Reply:
> I'm trying to figure out how to add the chaiscript .so's to the path

What is the reason of this exactly?

Reply:
For 'make test'

Reply:
> For 'make test'

I understand that. I mean what is the real reason behind it. Why this library should be located exactly in `${CMAKE_BINARY_DIR}`? Usually CMake manage such things as RPATH automatically and you don't have to do any extra customization about it. Why default behavior is not working? On what platforms do you have problems?